### PR TITLE
Backend: Sulphur Skitter performance updates

### PR DESCRIPTION
**Pull request by hannibal002**
## What
optimizing Sulphur skitter box

## Changelog Technical Details
+ Optimized sulphur skitter performance by caching `AxisAlignedBB` calculations and using more `LorenzVec`. - hannibal2

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/nether/SulphurSkitterBox.kt [hunk 1]
```diff
@@ -26,54 +27,63 @@ import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
...
+            val pos1 = it.add(-RADIUS, -RADIUS, -RADIUS)
+            val pos2 = it.add(RADIUS + 1, RADIUS + 1, RADIUS + 1)
+            pos1.axisAlignedTo(pos2).expandBlock()
         }
     }
 
+    private fun calculateSpongeLocations() {
+        val location = LocationUtils.playerLocation()
+        val from = location.add(-20, -20, -20).toBlockPos()
+        val to = location.add(20, 20, 20).toBlockPos()
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/nether/SulphurSkitterBox.kt]: if 2 lines down we check if its within 15 blocks why is it 20 blocks up here?
by changing these all to 15, we can check 39000 less blocks each time this function is ran (which is only 43% of the original blocks checked)

**Review by CalMWolfs — CHANGES_REQUESTED** [review]

File: src/main/java/at/hannibal2/skyhanni/features/nether/SulphurSkitterBox.kt (hunk 1 above)
Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/features/nether/SulphurSkitterBox.kt]: Those numbers aren't from me.
I assume the reason is so blocks in a 15 block radius will get instantly detected and not only every second

Comment (CalMWolfs) [reply src/main/java/at/hannibal2/skyhanni/features/nether/SulphurSkitterBox.kt]: > Those numbers aren't from me. I assume the reason is so blocks in a 15 block radius will get instantly detected and not only every second

Those numbers may not be from you but this is a performance improvement pr and i have given performance improvement suggestion

Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/features/nether/SulphurSkitterBox.kt]: I agree with the math though, we can set both to 15

**Review by CalMWolfs — APPROVED** [review]

**Pull request merged**